### PR TITLE
statistics, util/ranger: add cardinality estimation trace for `GetRowCountBy...`

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: TODO

### What is changed and how it works?

Add cardinality tracing framework and tracing for `GetRowCountBy...()`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: statistics/table.go [hunk 1]
```diff
@@ -331,39 +334,65 @@ func (t *Table) ColumnEqualRowCount(sc *stmtctx.StatementContext, value types.Da
...
-		return GetPseudoRowCountByColumnRanges(sc, float64(coll.Count), colRanges, 0)
+		result, err := GetPseudoRowCountByColumnRanges(sc, float64(coll.Count), colRanges, 0)
+		if err == nil && sc.EnableOptimizerCETrace && ok {
+			CETraceRange(sc, coll.PhysicalID, []string{c.Info.Name.O}, colRanges, "Column Stats-Pseudo", uint64(result))
+		}
+		return result, err
 	}
 	result, err := c.GetColumnRowCount(sc, colRanges, coll.Count, false)
+	if sc.EnableOptimizerCETrace {
+		CETraceRange(sc, coll.PhysicalID, []string{c.Info.Name.O}, colRanges, "Column Stats", uint64(result))
```
Comment (qw4990) [inline statistics/table.go:372]: `Column Stats` is not unified with `Int Column Stats`, better to use `Nonint Column Stats` or just use `Column Stats` in both places.

**Review by qw4990 — APPROVED** [review]

File: statistics/table.go (hunk 1 above)
Comment (time-and-fate) [reply statistics/table.go:372]: Updated.

**Review by Yisaer — APPROVED** [review]

**Pull request merged**